src/test/java/org/opentripplanner/sandbox/SandboxConfigurationTest.java

```diff
@@ -0,0 +1,75 @@
+package org.opentripplanner.sandbox;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+public class SandboxConfigurationTest {
+    private static final String FEATURE_A = "Setup feature A.";
+    private static final String FEATURE_B_PREFIX = "Setup feature B with profile: ";
+    private static final String PROFILE = "Norway";
+    private static final String JSON_CONFIG = (""
+            + "{ "
+            + "   'extensions': {"
+            + "       'feature-A' : true, "
+            + "       'feature-B': { "
+            + "           'profile' : '" + PROFILE + "'"
+            + "       }"
+            + "   } "
+            + "}").replace('\'', '\"');
+
+    private static final List<String> results = new ArrayList<>();
+
+
+    @Test
+    public void testSetup() throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+
+        //objectMapper.
+
+        JsonNode node = objectMapper.readTree(JSON_CONFIG);
+
+
+        SandboxConfiguration config = new SandboxConfiguration(node);
+
+        config.addTestExtension("feature-A", FeatureA.class);
+        config.addTestExtension("feature-B", FeatureB.class);
+
+        config.setup();
+
+        assertTrue("Result contain feature A: " + results, results.contains(FEATURE_A));
+        assertTrue("Result contain feature B: " + results, results.contains(resultFeatureB(PROFILE)));
+    }
+
+
+    static class FeatureA implements SandboxExtensionConfiguration {
+        public void setup() {
+            results.add(FEATURE_A);
+        }
+    }
+
+    static class FeatureB implements SandboxExtensionConfiguration {
+        private String profile;
```

Comment:
As illustrated in this test, I have chosen to inject config from JSON directly on the `SandboxExtensionConfiguration ` element it self - this is convenient, but might get in the way if the configuration is complex?

Any thoughts on this?

Reply:
The alternative is to pass the `JsonNode` to the `SandboxExtensionConfiguration#setup` method, but I think it is unnecessary for all extensions to implement the parsing of the config them selves. Also, the way I did it make it consistent - and we could extract this code and make all configuration in OTP work this way, enabling som automatic testing of the doc - possible autogenerating the configuration documentation from code.